packages/apps/reputation-oracle/server/src/modules/slack/slack.service.ts

```diff
@@ -0,0 +1,169 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+import { firstValueFrom } from 'rxjs';
+import { SlackConfigService } from '../../config/slack-config.service';
+import logger from '../../logger';
+import { AbuseDecision } from '../abuse/constants';
+import { ChainId, EscrowUtils, OperatorUtils } from '@human-protocol/sdk';
+
+@Injectable()
+export class SlackService {
+  private readonly logger = logger.child({ context: SlackService.name });
+
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly slackConfigService: SlackConfigService,
+  ) {}
+
+  private async sendNotification(message: any): Promise<void> {
+    try {
+      await firstValueFrom(
+        this.httpService.post(this.slackConfigService.webhookUrl, message),
+      );
+    } catch (error) {
+      this.logger.error('Error sending Slack notification:', error);
+      throw error;
+    }
+  }
+
+  private async openModal(triggerId: string, modalView: any): Promise<void> {
+    try {
+      const response = await firstValueFrom(
+        this.httpService.post(
+          'https://slack.com/api/views.open',
+          {
+            trigger_id: triggerId,
+            view: modalView,
+          },
+          {
+            headers: {
+              Authorization: `Bearer ${this.slackConfigService.oauthToken}`,
+              'Content-Type': 'application/json',
+            },
+          },
+        ),
+      );
+
+      if (!response.data.ok) {
+        this.logger.error('Error opening Slack modal:', response.data);
+        throw new Error('Failed to open Slack modal');
+      }
+    } catch (error) {
+      this.logger.error('Error opening Slack modal:', error);
+      throw error;
+    }
+  }
+
+  async updateMessage(responseUrl: string, text: string): Promise<void> {
+    try {
+      await firstValueFrom(this.httpService.post(responseUrl, { text }));
+    } catch (error) {
+      this.logger.error('Error updating Slack message:', error);
+      throw error;
+    }
+  }
+
+  async sendAbuseNotification(data: {
+    abuseId: number;
+    chainId: ChainId;
+    escrowAddress: string;
+  }): Promise<void> {
+    const escrow = await EscrowUtils.getEscrow(
```

Comment:
This method should accept all necessary data for the message and not have any system-specific business logic, i.e. just take some input, format it to slack message/modal/whatever and send